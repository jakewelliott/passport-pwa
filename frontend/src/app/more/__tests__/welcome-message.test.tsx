import WelcomeMessage from '@/app/more/welcome-message';
import { render, screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

describe('WelcomeMessage Component', () => {
  it('renders the title correctly', () => {
    render(<WelcomeMessage />);

    const title = screen.getByText('WELCOME!');
    expect(title).toBeInTheDocument();
    expect(title).toHaveClass('break-words', 'text-center', 'text-main_green');
  });

  it('renders all paragraphs correctly', () => {
    render(<WelcomeMessage />);

    const paragraphs = [
      "Since 1916, North Carolina's state parks have grown a lot. Now they cover over a 260,000 acres and offer tons of things to do and see. There's something for everyone!",
      'You can find all sorts of landscapes, from tall forests to calm lakes and winding rivers. Plus, there are miles of trails for hiking, mountain biking, and horseback riding. You can also paddle through water routes, enjoy picnics in beautiful spots, and spend a night under the stars.',
      "These parks are perfect for taking a break, connecting with nature, and feeling refreshed. You'll discover amazing places that show how we fit into the natural world. There are also exhibits and programs to learn more about North Carolina's wonders.",
      "We challenge you to visit as many of these amazing places as you can. Come join us to explore and learn about North Carolina's awesome natural and cultural treasures. We hope this program inspires you to create unforgettable memories and capture keepsakes of your adventures.",
    ];

    for (const text of paragraphs) {
      const paragraph = screen.getByText(text);
      expect(paragraph).toBeInTheDocument();
    }
  });

  it('renders all images with correct attributes', () => {
    render(<WelcomeMessage />);

    const images = [
      {
        src: '/photos/ELKN_ParkVisitor_LauraMeeks.jpg',
        alt: 'Elk Knob State Park Scene',
      },
      {
        src: '/photos/MOJESign_ParkVisitor_EverettFamily.jpg',
        alt: 'Mount Jefferson State Natural Area Scene',
      },
      {
        src: '/photos/DogWithPassport_ParkVisitor_KatieGriswold.jpg',
        alt: 'Dog with paper Passport',
      },
    ];

    for (const { src, alt } of images) {
      const image = screen.getByAltText(alt);
      expect(image).toBeInTheDocument();
      expect(image).toHaveAttribute('src', src);
      expect(image).toHaveClass('aspect-square', 'w-full', 'rounded-lg', 'object-cover', 'shadow-md');
    }
  });

  it('renders the photo caption correctly', () => {
    render(<WelcomeMessage />);

    const caption = screen.getByText(
      'From left to right - photo courtesy of: Laura Meeks; The Everett Family; Katie Griswold',
    );
    expect(caption).toBeInTheDocument();
    expect(caption).toHaveClass('p-mini');
    expect(caption).toHaveStyle({ marginTop: '12px' });
  });

  it('renders the layout structure correctly', () => {
    render(<WelcomeMessage />);

    // Check main container
    const mainContainer = screen.getByText('WELCOME!').closest('div');
    expect(mainContainer).toHaveClass('mx-auto', 'max-w-3xl', 'space-y-6', 'p-6');

    // Check image grid container
    const imageGrid = screen.getByTestId('image-grid');
    expect(imageGrid).toBeInTheDocument();
    expect(imageGrid).toHaveClass('mt-8', 'grid', 'grid-cols-3', 'place-items-center', 'gap-4');
  });
});
