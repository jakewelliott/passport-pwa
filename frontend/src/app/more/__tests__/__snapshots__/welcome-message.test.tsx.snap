// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WelcomeMessage Component > matches snapshot 1`] = `
<div>
  <div
    class="space-y-6"
  >
    <h1
      class="whitespace-nowrap text-center text-main-green text-xl"
    >
      <span
        class="inline-block w-full text-[length:min(64px,_calc(100vw_/_7))]"
      >
        WELCOME!
      </span>
    </h1>
    <div
      class="space-y-4"
    >
      <p>
        Since 1916, North Carolina's state parks have grown a lot. Now they cover over a 260,000 acres and offer tons of things to do and see. There's something for everyone!
      </p>
      <p>
        You can find all sorts of landscapes, from tall forests to calm lakes and winding rivers. Plus, there are miles of trails for hiking, mountain biking, and horseback riding. You can also paddle through water routes, enjoy picnics in beautiful spots, and spend a night under the stars.
      </p>
      <p>
        These parks are perfect for taking a break, connecting with nature, and feeling refreshed. You'll discover amazing places that show how we fit into the natural world. There are also exhibits and programs to learn more about North Carolina's wonders.
      </p>
      <p>
        We challenge you to visit as many of these amazing places as you can. Come join us to explore and learn about North Carolina's awesome natural and cultural treasures. We hope this program inspires you to create unforgettable memories and capture keepsakes of your adventures.
      </p>
    </div>
    <div
      class="mt-8 grid grid-cols-3 place-items-center gap-4"
      data-testid="image-grid"
    >
      <div>
        <img
          alt="Elk Knob State Park Scene"
          class="aspect-square w-full rounded-lg object-cover shadow-md"
          src="/photos/ELKN_ParkVisitor_LauraMeeks.jpg"
        />
      </div>
      <div>
        <img
          alt="Mount Jefferson State Natural Area Scene"
          class="aspect-square w-full rounded-lg object-cover shadow-md"
          src="/photos/MOJESign_ParkVisitor_EverettFamily.jpg"
        />
      </div>
      <div>
        <img
          alt="Dog with paper Passport"
          class="aspect-square w-full rounded-lg object-cover shadow-md"
          src="/photos/DogWithPassport_ParkVisitor_KatieGriswold.jpg"
        />
      </div>
    </div>
    <p
      class="text-xs "
      style="margin-top: 12px;"
    >
      From left to right - photo courtesy of: Laura Meeks; The Everett Family; Katie Griswold
    </p>
  </div>
</div>
`;
